maayanlab_bioinformatics/enrichment/gsea2015.py

```diff
@@ -0,0 +1,23 @@
+import numpy as np
+import pandas as pd
+
+def GSEA2015(geneset_membership: pd.Series, correlations: pd.Series):
+  '''
+  Implementation of algorithm described here:
+  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1239896/
+
+  :param geneset_membership: (pd.Series) True if in set, False if not, index: all genes
+  :param correlations: (pd.Series) Correlation of a given gene
+  :return (Tuple[np.array, np.array]) x and y arrays ready to be plotted. ES = y.max()
+  '''
```

Comment:
I added a function which follows the paper in implementation and with notation for clarity